### PR TITLE
Fake checkpoint info

### PR DESCRIPTION
**Pull request by wkpark**
This hack fixes some issues.
- [x] fix output file name - you can use `[seed]-[model_name]` in the settings -> Saving Images/grids -> Images filename pattern
- [x] fix cache issue with the `sd_checkpoint_cache > 0` setting
- [x] in this hack, `sha256` is a fake hash, (fake hash is ~~`sha256(currentmodel)`~~ `sha256 = hashlib.sha256(json.dumps(metadata).encode("utf-8")).hexdigest()`. and it is used in the pnginfo as a `Model hash:` )
- [x] check save work correctly.
- [x] using sd-webui's `load_model()` solve cuda memory issue

currently, `currentmodel` is used for "`model_name`". (`checkpoint_info.name_for_extra` is used for `model_name` internally)

`Model hash:` could be replaced/fixed using `script_callbacks.on_before_image_saved(my_image_fixer)`

### CODE REVIEW

**Comment by hako-mikan** [comment]
In this approach, wouldn't the merged model still be allocated to the cache? 
So you're saying that if we're merging two models consecutively, we would need to have three caches, correct?

**Comment by wkpark** [comment]
> In this approach, wouldn't the merged model still be allocated to the cache?

yes. right.

> So you're saying that if we're merging two models consecutively, we would need to have three caches, correct?

increase the cache size or not it will work without problem.

anyway, I've checked with different `sd_checkpoint_cache > 0` settings

~~~
Loading weights [Anything_v3Fixed-prunedFp16] from file
Loading weights [bc2f30f4ad] from cache
Creating model from config: F:\webui\webui\stable-diffusion-webui\configs\v1-inference.yaml
Loading VAE weights specified in settings: F:\webui\webui\stable-diffusion-webui\models\VAE\vae-ft-mse-840000-ema-pruned.safetensors
Applying attention optimization: sdp... done.
Model loaded in 2.3s (create model: 0.7s, apply weights to model: 1.1s, load VAE: 0.2s).
Loading weights [Brav6] from file
~~~

and newly merged models with the same settings have different `checkpoint_info`. -> results cached `checkpoints_loaded` separatly.
---------
call `sd_models.load_model(checkpoint_info)` with `already_loaded_state_dict=state_dict` args mess up `checkpoints_loaded` and `checkpoint_aliases` caches. simply use `deepcopy(checkpoint_info)` remove this issue.

- `checkpoints_loaded` use `checkpoint_info` as a key. `deepcopy()` remove this issue. (after call `load_model()`, `checkpints_loaded[checkpoint_info]` is changed with a given merged `state_dict`. That's not what we want)

**Comment by hako-mikan** [comment]
I also think that using sd_models.load_model(checkpoint_info) with already_loaded_state_dict=state_dict would be a better approach, especially when considering future updates of web-ui. 
The issue is that the required cache size increases when merging two models consecutively. 
If we're merging SDXL models, 64GB of RAM might not be sufficient, which could pose a problem for some users.

Perhaps a more suitable solution might be to create a Pull Request to add a "no-cache" option to sd_models.load_model(checkpoint_info).

The reason I created this script is precisely because file read-write operations were taking too long. A decrease in processing speed due to the necessity of reading from the disk, when cache is insufficient, is not what I am aiming for.

I'll give this issue some more thought as well.

**Comment by wkpark** [comment]
I guess @hako-mikan found another way to workaround the cache related issue.
Thanks for your great work!

**Comment by hako-mikan** [comment]
The caching functionality on the web-ui side stopped working properly for supermerger, so I decided to implement a new cache on our end. 
This has enabled performance improvements and the use of 'load_model'."
